TiFBackendUtils/SQLExecutable/utils.ts

```diff
@@ -10,19 +10,47 @@ import { AwaitableResult, failure, promiseResult, success } from "../result.js"
  * transaction or not.
  */
 
-type QueryResult = {
+type ExecuteResult = {
   insertId: string
   rowsAffected: number
 }
 
+const typecasts: Record<number, (value: string | null) => unknown> = {
+  3: (value) => parseInt(value ?? "0") > 0, // INT or LONG as BOOLEAN
+  1: (value) => parseInt(value ?? "0") > 0, // TINYINT as BOOLEAN
+  12: (value) => value ? new Date(value) : value, // DATETIME
+  246: (value) => value ? parseFloat(value) : value // NEWDECIMAL (DECIMAL/NUMERIC)
+}
```

Comment:
Nit: Maybe create an `MYSQL_TYPE_ID` object to store these numerical keys. It would eliminate the need for the comments.

Reply:
I'll add in a follow up